pkg/cloudprovider/plugins/internal/cloud_common.go

```diff
@@ -202,7 +205,9 @@ func (c *cloudCommon) DoInventoryPoll(accountNamespacedName *types.NamespacedNam
 	}
 	accCfg.LockMutex()
 	defer accCfg.UnlockMutex()
-
+	if !accCfg.GetAccountConfigState() {
```

Comment:
Could you just update GetCloudAccountByName routine to return 'not found' or 'not valid' error.

Reply:
Updated the function to return, account config, found and an error.